File: Audio Inpainting/utils/evaluation/SNRInpaintingPerformance.py
import numpy as np
from snr import SNR
from typing import List, Tuple

def SNRInpaintingPerformance(
    xRef: np.ndarray, xObs: np.ndarray, 
    xEst: np.ndarray, IMiss: np.ndarray, 
    DISP_FLAG: int = 0) -> Tuple[List[float], List[float]]:
    """
    Compute various SNR measures for inpainting performance.

    Args:
        xRef (np.ndarray): Reference signal.
        xObs (np.ndarray): Observed signal.
        xEst (np.ndarray): Estimate signal.
        IMiss (np.ndarray): Location of missing data.
        DISP_FLAG (int, optional): Flag to display SNR on all samples/clipped samples. Defaults to 0.

    Returns:
        Tuple[List[float], List[float]]: A tuple of two lists.
            - SNRAll (List[float]): SNRAll[0] is the original SNR, between xRef and xObs; SNRAll[1] is the obtained SNR, between xRef and xEst.
            - SNRmiss (List[float]): The same as SNRAll but computed on the missing/restored samples only.
    """
    pass
    # Various SNR measures for inpainting performance
    SNRAll = [SNR(xRef, xObs), SNR(xRef, xEst)]
    SNRmiss = [SNR(xRef[IMiss], xObs[IMiss]), SNR(xRef[IMiss], xEst[IMiss])]

    if DISP_FLAG > 0:
        print('SNR on all samples / clipped samples:')
        print(f'Original: {SNRAll[0]} dB / {SNRmiss[0]} dB')
        print(f'Estimate: {SNRAll[1]} dB / {SNRmiss[1]} dB')

    return SNRAll, SNRmiss

File: AudioInpaintingToolbox/Utils/evaluation/SNRInpaintingPerformance.m
function [SNRAll,SNRmiss] = SNRInpaintingPerformance(xRef,xObs,xEst,IMiss,DISP_FLAG)
% Various SNR measures for inpainting performance
%
% Usage: [SNRAll,SNRmiss] = SNRInpaintingPerformance(xRef,xObs,xEst,IMiss,DISP_FLAG)
%
%
% Inputs:
%          - xRef - reference signal
%          - xObs - observed signal
%          - xEst - estimate signal
%          - IMiss - location of missing data
%
% Outputs:
%          - SNRAll - SNRAll(1) is the original SNR, between xRef and xObs; 
%          SNRAll(2) is the SNR is the obtained SNR, between xRef and xEst
%          - SNRmiss - the same as SNRAll but computed on the missing/restored
%          samples only
%
%
% -------------------
%
% Audio Inpainting toolbox
% Date: June 28, 2011
% By Valentin Emiya, Amir Adler, Maria Jafari
% This code is distributed under the terms of the GNU Public License version 3 (http://www.gnu.org/licenses/gpl.txt).
if nargin<5
   DISP_FLAG = 0;
end

SNRAll = [SNR(xRef,xObs),SNR(xRef,xEst)];
SNRmiss = [SNR(xRef(IMiss),xObs(IMiss)),SNR(xRef(IMiss),xEst(IMiss))];

if DISP_FLAG>0
   fprintf('SNR on all samples / clipped samples:\n');
   fprintf('Original: %g dB / %g dB\n',...
      SNRAll(1),...
      SNRmiss(1));
   fprintf('Estimate: %g dB / %g dB\n',...
      SNRAll(2),...
      SNRmiss(2));
end

return
